src/attacks.rs

```diff
@@ -14,15 +22,214 @@ pub enum DepthReduceSet {
+            DepthReduceSet::ValiantDepth(depth) => AttackTarget::Depth(depth as f64 / graph_size),
+            DepthReduceSet::ValiantSize(size) => AttackTarget::Size(size as f64 / graph_size),
+            DepthReduceSet::ValiantAB16(size) => AttackTarget::Size(size as f64 / graph_size),
+            DepthReduceSet::GreedyDepth(depth, _) => AttackTarget::Depth(depth as f64 / graph_size),
+            DepthReduceSet::GreedySize(size, _) => AttackTarget::Size(size as f64 / graph_size),
+        };
+        // FIXME: This code should absorb the `depth_reduce` and derived
+        // functions logic. The target discrimination depth/size should
+        // b independent of the attack type (valiant/greedy).
+
+        let range = {
+            let single_value = match target {
+                AttackTarget::Depth(depth) => depth,
+                AttackTarget::Size(size) => size,
+            };
+            // FIXME: Too verbose, there probably is a more concise way to do this.
+            TargetRange {
+                start: single_value,
+                end: single_value,
+                interval: 0.0,
+            }
+        };
+
+        AttackProfile {
+            runs: 1,
+            target,
+            range,
+            attack,
+        }
+    }
+}
+
+/// Results of an attack expressed in relation to the graph size.
+#[derive(Clone, Default, Debug)]
+pub struct SingleAttackResult {
+    depth: f64,
+    exclusion_size: f64,
+    // graph_size: usize,
+    // FIXME: Do we care to know the absolute number or just
+    // relative to the graph size?
+}
+
+impl<'a> std::iter::Sum<&'a Self> for SingleAttackResult {
+    fn sum<I>(iter: I) -> Self
+    where
+        I: Iterator<Item = &'a Self>,
+    {
+        iter.fold(SingleAttackResult::default(), |a, b| SingleAttackResult {
+            depth: a.depth + b.depth,
+            exclusion_size: a.exclusion_size + b.exclusion_size,
+        })
+    }
+}
+
+/// Average of many `SingleAttackResult`s.
+// FIXME: Should be turn into a more generalized structure that also
+//  has the variance along with the mean using a specialized crate.
+#[derive(Clone, Default, Debug)]
+pub struct AveragedAttackResult {
+    mean_depth: f64,
+    mean_size: f64,
+}
+
+impl AveragedAttackResult {
+    pub fn from_results(results: Vec<SingleAttackResult>) -> Self {
+        let aggregated: SingleAttackResult = results.iter().sum();
+        AveragedAttackResult {
+            mean_depth: aggregated.depth / results.len() as f64,
+            mean_size: aggregated.exclusion_size / results.len() as f64,
+        }
+    }
+}
+
+impl std::fmt::Display for SingleAttackResult {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        write!(
+            f,
+            "\t-> |S| = {:.4}\n\t-> depth(G-S) = {:.4}",
+            self.exclusion_size, self.depth,
+        )
+    }
+}
+
+pub fn attack(g: &mut Graph, attack: DepthReduceSet) -> SingleAttackResult {
```

Comment:
I would actually *keep* this method around if possible, it's very easy to setup and test this way